File: app/src/main/java/com/perflyst/twire/tasks/GetStreamChattersTask.java
package com.perflyst.twire.tasks;

import com.perflyst.twire.service.Service;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.concurrent.Callable;

/**
 * Created by Sebastian Rask on 22-01-2017.
 */

public class GetStreamChattersTask implements Callable<ArrayList<String>> {
    private final String mStreamTwitchName;

    public GetStreamChattersTask(String mStreamTwitchName) {
        this.mStreamTwitchName = mStreamTwitchName;
    }

    public ArrayList<String> call() {
        try {
            final String BASE_URL = "https://tmi.twitch.tv/group/user/" + mStreamTwitchName + "/chatters";

            JSONObject topObject = new JSONObject(Service.urlToJSONString(BASE_URL));

            return null;
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: app/src/main/java/com/perflyst/twire/tasks/GetStreamChattersTask.kt
package com.perflyst.twire.tasks

import com.perflyst.twire.service.Service
import org.json.JSONException
import org.json.JSONObject
import timber.log.Timber
import java.util.concurrent.Callable

/**
 * Created by Sebastian Rask on 22-01-2017.
 */
class GetStreamChattersTask(private val mStreamTwitchName: String) : Callable<ArrayList<String>?> {
    override fun call(): ArrayList<String>? {
        try {
            val BASE_URL = "https://tmi.twitch.tv/group/user/$mStreamTwitchName/chatters"

            val topObject = JSONObject(Service.urlToJSONString(BASE_URL))

            return null
        } catch (e: JSONException) {
            Timber.e(e)
        }
        return null
    }
}
